sqs/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/sinks/SqsSink.scala

```diff
@@ -136,38 +137,47 @@ class SqsSink private (
     }
   }
 
-  def sendBatch(batch: List[EventBuffer.Event], nextBackoff: Long = minBackoff): Unit =
+  def sinkBatch(batch: List[Events], nextBackoff: Long = minBackoff)(maxRetries: Int = MaxRetries): Unit =
     if (batch.nonEmpty) {
-      log.info(s"Writing ${batch.size} Thrift records to SQS queue ${queueName}")
+      log.info(s"Writing ${batch.size} Thrift records to SQS queue $queueName.")
 
-      multiPut(batch).onComplete {
+      writeBatchToSqs(batch).onComplete {
         case Success(s) =>
-          log.info(s"Successfully wrote ${batch.size - s.size} out of ${batch.size} messages.")
+          outage = false
+          log.info(s"Successfully wrote ${batch.size - s.size} out of ${batch.size} messages to SQS queue $queueName.")
           if (s.nonEmpty) {
             s.foreach { f =>
               log.error(
-                s"Message failed with error code [${f._2.code}] and message [${f._2.message}]"
+                s"Failed writing message to SQS queue $queueName, with error code [${f._2.code}] and message [${f._2.message}]."
               )
             }
-            log.error(s"Retrying in $nextBackoff milliseconds...")
-            scheduleBatch(s.map(_._1), nextBackoff)
+            log.error(
+              s"Retrying all messages that could not be written to SQS queue $queueName in $nextBackoff milliseconds..."
+            )
+            scheduleWrite(s.map(_._1), nextBackoff)(MaxRetries)
           }
         case Failure(f) =>
-          log.error("Writing failed.", f)
-          log.error(s"Retrying in $nextBackoff milliseconds...")
-          scheduleBatch(batch, nextBackoff)
+          log.error("Writing failed with error:", f)
+
+          if (maxRetries > 1) {
+            log.error(s"Retrying writing batch to SQS queue $queueName in $nextBackoff milliseconds...")
```

Comment:
A minor point - I would probably move the duplicated `log.error` line outside of the if-else block.